src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
If default formatting doesn't work, we should use FormattedSymbol class instead of prematurely producing strings. Please either fix this code and the code you used as a template, or open an issue to address this later.